crates/task-impls/src/consensus.rs

```diff
@@ -471,6 +471,34 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
         false
     }
 
+    /// Validates whether the VID Dispersal Proposal is correctly signed
+    fn validate_disperse(&self, disperse: &Proposal<TYPES, VidDisperseShare<TYPES>>) -> bool {
+        let view = disperse.data.get_view_number();
+        let payload_commitment = disperse.data.payload_commitment;
+        // Check whether the data comes from the right leader for this view
+        if self
+            .quorum_membership
+            .get_leader(view)
+            .validate(&disperse.signature, payload_commitment.as_ref())
+        {
+            return true;
+        }
+        // or the data was calculated and signed by the current node
+        if self
+            .public_key
+            .validate(&disperse.signature, payload_commitment.as_ref())
+        {
+            return true;
+        }
+        // or the data was signed by one of the staked DA committee members
+        for da_member in self.committee_membership.get_staked_committee(view) {
+            if da_member.validate(&disperse.signature, payload_commitment.as_ref()) {
+                return true;
+            }
+        }
```

Comment:
Do the DA members actually sign the disperse? I thought were were passing along the leaders signature, but that might have changed when you split it up?